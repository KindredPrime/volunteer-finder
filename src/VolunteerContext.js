import React from 'react';

const VolunteerContext = React.createContext({
  orgs: [],
  causes: [],
  addOrg: () => {},
  deleteOrg: () => {},
  appError: null,
  // Is the app fetching data from the API?
  fetching: false
});

export default VolunteerContext;