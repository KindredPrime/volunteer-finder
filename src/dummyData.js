const dummyUsers = [
  {
    id: 1,
    email: 'email1@email.com',
    username: 'user1',
    password: 'password1',
    orgsAdded: [1],
    eventsAdded: [1, 2]
  },
  {
    id: 2,
    email: 'email2@email.com',
    username: 'user2',
    password: 'password2',
    orgsAdded: [2],
    eventsAdded: [3]
  },
  {
    id: 3,
    email: 'email3@email.com',
    username: 'user3',
    password: 'password3',
    orgsAdded: [3],
    eventsAdded: []
  },
  {
    id: 4,
    email: 'email4@email.com',
    username: 'user4',
    password: 'password4',
    orgsAdded: [],
    eventsAdded: []
  },
  {
    id: 5,
    email: 'email5@email.com',
    username: 'user5',
    password: 'password5',
    orgsAdded: [4],
    eventsAdded: [4]
  }
];

const dummyOrgs = [
  {
    id: 1,
    name: 'YMCA',
    website: 'https://www.ymca.com',
    phone: '1-800-555-7777',
    email: 'contact@ymca.com',
    address: '1 YMCA Street Real City, Real State',
    causes: [1, 2, 3],
    tags: [2, 3, 4, 6]
  },
  {
    id: 2,
    name: 'ASPCA',
    website: 'https://www.aspca.com',
    phone: '1-800-123-4567',
    email: 'contact@aspca.com',
    address: '1 ASPCA Street Real City, Real State',
    causes: [5, 2],
    tags: [2, 3, 4, 10]
  },
  {
    id: 3,
    name: 'National VOAD',
    website: 'https://www.nvoad.com',
    phone: '1-800-456-7890',
    email: 'contact@nvoad.com',
    address: '1 NVOAD Street Rivendell, Middle Earth',
    causes: [4, 2],
    tags: [2, 3, 4, 5, 7, 11]
  },
  {
    id: 4,
    name: 'National Air and Space Museum',
    website: 'https://www.airandspace.com',
    phone: '1-800-987-6543',
    email: 'contact@airandspace.com',
    address: '1 Space Street Air City, Outer Space',
    causes: [6, 7],
    tags: [1, 2, 3, 4, 6, 12]
  }
];

const dummyEvents = [
  {
    id: 1,
    name: 'Member Appreciation Day',
    organization: 1,
    location: 'Reston YMCA Center',
    date: 'February 20, 2020 00:00:00',
    duration: '4 hours',
    description: 'Greet YMCA members as they enter the facility and offer them snacks.  See https://www.website.com for more details.',
    causes: [2, 3],
    tags: [2, 4, 12]
  },
  {
    id: 2,
    name: 'Triathlon Assistance',
    organization: 1,
    location: 'Reston YMCA Center',
    date: 'January 27, 2020 00:00:00',
    duration: '4 hours',
    description: 'Help YMCA staff host their annual indoor triathlon: sign in triathlon participants, direct the participants to the different stages of the triathlon, and record times. See https://www.triathlon.com for more details.',
    causes: [2, 3],
    tags: [2, 3, 9]
  },
  {
    id: 3,
    name: 'Adoption Day',
    organization: 2,
    location: 'Times Square, New York City, New York',
    date: 'December 13, 2020 00:00:00',
    duration: '8 hours',
    description: 'Help the ASPCA run their annual adoption day for their local pet shelter.',
    causes: [5],
    tags: [2, 3, 9, 12]
  },
  {
    id: 4,
    name: 'Docent Training Class',
    organization: 4,
    location: '14390 Air and Space Museum Pkwy Chantilly, VA 20151',
    date: 'January 30, 2021 00:00:00',
    duration: '2 hours',
    description: 'This is a required class for anyone who wants to be a docent (tour guide) at the National Air and Space Museum.',
    causes: [6, 7],
    tags: [2, 3, 13]
  }
];

const dummyTags = [
  {
    id: 1,
    name: 'virtual'
  },
  {
    id: 2,
    name: 'in-person'
  },
  {
    id: 3,
    name: 'weekends'
  },
  {
    id: 4,
    name: 'weekdays'
  },
  {
    id: 5,
    name: 'nights'
  },
  {
    id: 6,
    name: 'group'
  },
  {
    id: 7,
    name: 'delivery'
  },
  {
    id: 8,
    name: 'cooking'
  },
  {
    id: 9,
    name: 'administration'
  },
  {
    id: 10,
    name: 'cleaning'
  },
  {
    id: 11,
    name: 'heavy lifting'
  },
  {
    id: 12,
    name: 'greeting'
  },
  {
    id: 13,
    name: 'training'
  }
];

const dummyCauses = [
  {
    id: 1,
    name: 'Youth'
  },
  {
    id: 2,
    name: 'Community'
  },
  {
    id: 3,
    name: 'Health'
  },
  {
    id: 4,
    name: 'Disaster Response'
  },
  {
    id: 5,
    name: 'Animals'
  },
  {
    id: 6,
    name: 'Museums'
  },
  {
    id: 7,
    name: 'Education'
  },
  {
    id: 8,
    name: 'Human Rights'
  },
  {
    id: 9,
    name: 'Arts'
  }
];

export {
  dummyUsers,
  dummyOrgs,
  dummyEvents,
  dummyTags,
  dummyCauses
};