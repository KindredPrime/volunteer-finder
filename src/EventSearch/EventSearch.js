import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { updateField, getEntitiesById, setCheckboxValue } from '../util';
import VolunteerContext from '../VolunteerContext';
import Nav from '../Nav/Nav';
import EntityCheckboxes from '../EntityCheckboxes/EntityCheckboxes';
import SearchResults from '../SearchResults/SearchResults';

class EventSearch extends Component {
  static contextType = VolunteerContext;

  static defaultProps = {
    pageLimit: 10
  };

  // Only causes and tags that are checked are kept in the state
  state = {
    searchTerm: '',
    orgs: {

    },
    causes: {

    },
    tags: {

    },
    searchResults: [],
    page: 1,
    searched: false
  };

  handleSubmit = (e) => {
    e.preventDefault();

    const { searchTerm } = this.state;
    const searchOrgs = Object.entries(this.state.orgs);
    const searchCauses = Object.entries(this.state.causes);
    const searchTags = Object.entries(this.state.tags);

    const allOrgs = this.context.orgs;
    const allCauses = this.context.causes;
    const allTags = this.context.tags;
    const allEvents = this.context.events;

    const searchResults = allEvents
      .map((event) => ({
        ...event,
        organization: getEntitiesById([event.organization], allOrgs)[0],
        causes: getEntitiesById(event.causes, allCauses),
        tags: getEntitiesById(event.tags, allTags)
      }))
      .filter((event) => {
        const { name, location, description } = event;
        const eventOrg = event.organization;
        const eventCauses = event.causes;
        const eventTags = event.tags;
        const regEx = new RegExp(searchTerm, 'i');

        return (regEx.test(name) || regEx.test(eventOrg) || regEx.test(location) || regEx.test(description)
          )
          && (searchOrgs.length === 0 || searchOrgs.find(([org, __]) => org === eventOrg.name))
          && (searchCauses.length === 0 || eventCauses
            .find((eventCause) => searchCauses.find(([cause, __]) => (
              cause === eventCause.name
            ))))
          && (searchTags.length === 0 || eventTags
            .find((eventTag) => searchTags.find(([tag, __]) => ( 
              tag === eventTag.name
            ))));
      });

    this.setState({
      searchResults,
      searched: true
    });
  }

  render() {
    // Every time the component is rendered, it grabs the current context for the app's causes and 
    //  tags, to render dynamically on the page.
    const { orgs, causes, tags } = this.context;
    const { searchResults, searched } = this.state;
    const { pageLimit } = this.props;

    return (
      <div className="EventSearch">
        <Nav />
        
        <header>
          <h1>Search for Events</h1>
        </header>

        <form onSubmit={this.handleSubmit}>
          <div>
            <label htmlFor="search-term">Search Term</label>
            <input 
              type="text"
              id="search-term"
              onChange={(e) => updateField('searchTerm', e.target.value, this)}
            />
          </div>

          <br />

          {orgs && orgs.length > 0 && (
            <>
              <EntityCheckboxes 
                entities={orgs} handleClick={setCheckboxValue('orgs', this)}
                type="organizations"
              />

              <br />
            </>
          )}

          {causes && causes.length > 0 && (
            <>
              <EntityCheckboxes 
                entities={causes} handleClick={setCheckboxValue('causes', this)} 
                type="causes" 
              />

              <br />
            </>
          )}

          {tags && tags.length > 0 && (
            <EntityCheckboxes 
              entities={tags} handleClick={setCheckboxValue('tags', this)} 
              type="tags" 
            />
          )}

          <button
            type="submit"
          >
            Search
          </button>
        </form>

        {searched && <SearchResults results={searchResults} pageLimit={pageLimit} />}
      </div>
    );
  }
}

EventSearch.propTypes = {
  pageLimit: PropTypes.number.isRequired
};

export default EventSearch;