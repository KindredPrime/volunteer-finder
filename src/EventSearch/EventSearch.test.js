import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import EventSearch from './EventSearch';
import { dummyOrgs, dummyEvents, dummyCauses, dummyTags } from '../dummyData';
import VolunteerContext from '../VolunteerContext';

describe('EventSearch Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <EventSearch />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const contextValue = {
      orgs: dummyOrgs,
      causes: dummyCauses,
      tags: dummyTags
    };
    render(
      <VolunteerContext.Provider value={contextValue}>
        <BrowserRouter>
          <EventSearch />
        </BrowserRouter>
      </VolunteerContext.Provider>
    );

    expect(document.body).toMatchSnapshot();
  });

  it(`renders search results after clicking 'Search'`, () => {
    const contextValue = {
      orgs: dummyOrgs,
      events: dummyEvents,
      causes: [],
      tags: []
    };
    render(
      <VolunteerContext.Provider value={contextValue}>
        <BrowserRouter>
          <EventSearch pageLimit={7} />
        </BrowserRouter>
      </VolunteerContext.Provider>
    );

    userEvent.click(screen.getByRole('button', { name: 'Search' }));
    expect(document.body).toMatchSnapshot();
  });

  it('renders search results that match the search term', () => {
    const contextValue = {
      orgs: [],
      events: dummyEvents,
      causes: [],
      tags: []
    };
    render(
      <VolunteerContext.Provider value={contextValue}>
        <BrowserRouter>
          <EventSearch pageLimit={10} />
        </BrowserRouter>
      </VolunteerContext.Provider>
    );

    userEvent.type(screen.getByLabelText('Search Term'), 'YMCA');
    userEvent.click(screen.getByRole('button', { name: 'Search' }));
    expect(document.body).toMatchSnapshot();
  });

  it('renders search results that have the selected orgs', () => {
    const contextValue = {
      orgs: dummyOrgs.slice(4, 5),
      events: dummyEvents,
      causes: [],
      tags: []
    };
    render(
      <VolunteerContext.Provider value={contextValue}>
        <BrowserRouter>
          <EventSearch pageLimit={10} />
        </BrowserRouter>
      </VolunteerContext.Provider>
    );

    userEvent.click(screen.getByLabelText('Fairfax County Health Department'));
    userEvent.click(screen.getByRole('button', { name: 'Search' }));
    expect(document.body).toMatchSnapshot();
  });

  it('renders search results that have the selected causes', () => {
    const contextValue = {
      orgs: [],
      events: dummyEvents,
      causes: dummyCauses.slice(0, 1),
      tags: []
    };
    render(
      <VolunteerContext.Provider value={contextValue}>
        <BrowserRouter>
          <EventSearch pageLimit={10} />
        </BrowserRouter>
      </VolunteerContext.Provider>
    );

    userEvent.click(screen.getByLabelText('Youth'));
    userEvent.click(screen.getByRole('button', { name: 'Search' }));
    expect(document.body).toMatchSnapshot();
  });

  it('renders search results that have the selected tags', () => {
    const contextValue = {
      orgs: [],
      events: dummyEvents,
      causes: [],
      tags: dummyTags.slice(0, 1)
    };
    render(
      <VolunteerContext.Provider value={contextValue}>
        <BrowserRouter>
          <EventSearch pageLimit={10} />
        </BrowserRouter>
      </VolunteerContext.Provider>
    );

    userEvent.click(screen.getByLabelText('virtual'));
    userEvent.click(screen.getByRole('button', { name: 'Search' }));
    expect(document.body).toMatchSnapshot();
  });
});