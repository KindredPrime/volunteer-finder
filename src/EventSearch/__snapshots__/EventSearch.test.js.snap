// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventSearch Component renders search results after clicking 'Search' 1`] = `
<body>
  <div>
    <div
      class="EventSearch"
    >
      <nav
        class="Nav"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
        <a
          href="/org-search"
        >
          Search Organizations
        </a>
        <a
          href="/event-search"
        >
          Search Events
        </a>
        <a
          href="/login"
        >
          Login
        </a>
        <a
          href="/signup"
        >
          Sign Up
        </a>
      </nav>
      <header>
        <h1>
          Search for Events
        </h1>
      </header>
      <form>
        <div>
          <label
            for="search-term"
          >
            Search Term
          </label>
          <input
            id="search-term"
            type="text"
          />
        </div>
        <br />
        <fieldset
          class="EntityCheckboxes"
        >
          <legend>
            Organizations
             (all are selected by default)
          </legend>
          <div>
            <label
              for="organizations-1"
            >
              YMCA DC
            </label>
            <input
              id="organizations-1"
              name="organizations"
              type="checkbox"
              value="YMCA DC"
            />
          </div>
          <div>
            <label
              for="organizations-2"
            >
              Humane Society of Fairfax County, Inc
            </label>
            <input
              id="organizations-2"
              name="organizations"
              type="checkbox"
              value="Humane Society of Fairfax County, Inc"
            />
          </div>
          <div>
            <label
              for="organizations-3"
            >
              National VOAD
            </label>
            <input
              id="organizations-3"
              name="organizations"
              type="checkbox"
              value="National VOAD"
            />
          </div>
          <div>
            <label
              for="organizations-4"
            >
              National Air and Space Museum
            </label>
            <input
              id="organizations-4"
              name="organizations"
              type="checkbox"
              value="National Air and Space Museum"
            />
          </div>
          <div>
            <label
              for="organizations-5"
            >
              Fairfax County Health Department
            </label>
            <input
              id="organizations-5"
              name="organizations"
              type="checkbox"
              value="Fairfax County Health Department"
            />
          </div>
          <div>
            <label
              for="organizations-6"
            >
              Sully Historic Site
            </label>
            <input
              id="organizations-6"
              name="organizations"
              type="checkbox"
              value="Sully Historic Site"
            />
          </div>
          <div>
            <label
              for="organizations-7"
            >
              Vecinos Unidos
            </label>
            <input
              id="organizations-7"
              name="organizations"
              type="checkbox"
              value="Vecinos Unidos"
            />
          </div>
          <div>
            <label
              for="organizations-8"
            >
              Girls on the Run of Northern Virginia
            </label>
            <input
              id="organizations-8"
              name="organizations"
              type="checkbox"
              value="Girls on the Run of Northern Virginia"
            />
          </div>
          <div>
            <label
              for="organizations-9"
            >
              Optimist Club of Herndon, VA
            </label>
            <input
              id="organizations-9"
              name="organizations"
              type="checkbox"
              value="Optimist Club of Herndon, VA"
            />
          </div>
          <div>
            <label
              for="organizations-10"
            >
              FACETS
            </label>
            <input
              id="organizations-10"
              name="organizations"
              type="checkbox"
              value="FACETS"
            />
          </div>
          <div>
            <label
              for="organizations-11"
            >
              American Cancer Society in District of Columbia
            </label>
            <input
              id="organizations-11"
              name="organizations"
              type="checkbox"
              value="American Cancer Society in District of Columbia"
            />
          </div>
        </fieldset>
        <br />
        <button
          type="submit"
        >
          Search
        </button>
      </form>
      <section
        class="SearchResults"
      >
        <header>
          <h2>
            Results
          </h2>
        </header>
        <ul>
          <li>
            <a
              href="/event/1"
            >
              Member Appreciation Day
            </a>
          </li>
          <li>
            <a
              href="/event/2"
            >
              Triathlon Assistance
            </a>
          </li>
          <li>
            <a
              href="/event/3"
            >
              Adoption Day
            </a>
          </li>
          <li>
            <a
              href="/event/4"
            >
              Docent Training Class
            </a>
          </li>
          <li>
            <a
              href="/event/5"
            >
              Touring Docent Sully Historic Site
            </a>
          </li>
          <li>
            <a
              href="/event/6"
            >
              Volunteer Drivers needed to assist older adults in the community
            </a>
          </li>
          <li>
            <a
              href="/event/7"
            >
              Help Children with Homework and Reading
            </a>
          </li>
        </ul>
        <div
          class="SearchResults__page-buttons"
        >
          <button
            disabled=""
            type="button"
          >
            Previous
          </button>
          <button
            type="button"
          >
            Next
          </button>
        </div>
      </section>
    </div>
  </div>
</body>
`;

exports[`EventSearch Component renders the UI as expected 1`] = `
<body>
  <div>
    <div
      class="EventSearch"
    >
      <nav
        class="Nav"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
        <a
          href="/org-search"
        >
          Search Organizations
        </a>
        <a
          href="/event-search"
        >
          Search Events
        </a>
        <a
          href="/login"
        >
          Login
        </a>
        <a
          href="/signup"
        >
          Sign Up
        </a>
      </nav>
      <header>
        <h1>
          Search for Events
        </h1>
      </header>
      <form>
        <div>
          <label
            for="search-term"
          >
            Search Term
          </label>
          <input
            id="search-term"
            type="text"
          />
        </div>
        <br />
        <fieldset
          class="EntityCheckboxes"
        >
          <legend>
            Organizations
             (all are selected by default)
          </legend>
          <div>
            <label
              for="organizations-1"
            >
              YMCA DC
            </label>
            <input
              id="organizations-1"
              name="organizations"
              type="checkbox"
              value="YMCA DC"
            />
          </div>
          <div>
            <label
              for="organizations-2"
            >
              Humane Society of Fairfax County, Inc
            </label>
            <input
              id="organizations-2"
              name="organizations"
              type="checkbox"
              value="Humane Society of Fairfax County, Inc"
            />
          </div>
          <div>
            <label
              for="organizations-3"
            >
              National VOAD
            </label>
            <input
              id="organizations-3"
              name="organizations"
              type="checkbox"
              value="National VOAD"
            />
          </div>
          <div>
            <label
              for="organizations-4"
            >
              National Air and Space Museum
            </label>
            <input
              id="organizations-4"
              name="organizations"
              type="checkbox"
              value="National Air and Space Museum"
            />
          </div>
          <div>
            <label
              for="organizations-5"
            >
              Fairfax County Health Department
            </label>
            <input
              id="organizations-5"
              name="organizations"
              type="checkbox"
              value="Fairfax County Health Department"
            />
          </div>
          <div>
            <label
              for="organizations-6"
            >
              Sully Historic Site
            </label>
            <input
              id="organizations-6"
              name="organizations"
              type="checkbox"
              value="Sully Historic Site"
            />
          </div>
          <div>
            <label
              for="organizations-7"
            >
              Vecinos Unidos
            </label>
            <input
              id="organizations-7"
              name="organizations"
              type="checkbox"
              value="Vecinos Unidos"
            />
          </div>
          <div>
            <label
              for="organizations-8"
            >
              Girls on the Run of Northern Virginia
            </label>
            <input
              id="organizations-8"
              name="organizations"
              type="checkbox"
              value="Girls on the Run of Northern Virginia"
            />
          </div>
          <div>
            <label
              for="organizations-9"
            >
              Optimist Club of Herndon, VA
            </label>
            <input
              id="organizations-9"
              name="organizations"
              type="checkbox"
              value="Optimist Club of Herndon, VA"
            />
          </div>
          <div>
            <label
              for="organizations-10"
            >
              FACETS
            </label>
            <input
              id="organizations-10"
              name="organizations"
              type="checkbox"
              value="FACETS"
            />
          </div>
          <div>
            <label
              for="organizations-11"
            >
              American Cancer Society in District of Columbia
            </label>
            <input
              id="organizations-11"
              name="organizations"
              type="checkbox"
              value="American Cancer Society in District of Columbia"
            />
          </div>
        </fieldset>
        <br />
        <fieldset
          class="EntityCheckboxes"
        >
          <legend>
            Causes
             (all are selected by default)
          </legend>
          <div>
            <label
              for="causes-1"
            >
              Youth
            </label>
            <input
              id="causes-1"
              name="causes"
              type="checkbox"
              value="Youth"
            />
          </div>
          <div>
            <label
              for="causes-2"
            >
              Health
            </label>
            <input
              id="causes-2"
              name="causes"
              type="checkbox"
              value="Health"
            />
          </div>
          <div>
            <label
              for="causes-3"
            >
              Disaster Response
            </label>
            <input
              id="causes-3"
              name="causes"
              type="checkbox"
              value="Disaster Response"
            />
          </div>
          <div>
            <label
              for="causes-4"
            >
              Animals
            </label>
            <input
              id="causes-4"
              name="causes"
              type="checkbox"
              value="Animals"
            />
          </div>
          <div>
            <label
              for="causes-5"
            >
              History
            </label>
            <input
              id="causes-5"
              name="causes"
              type="checkbox"
              value="History"
            />
          </div>
          <div>
            <label
              for="causes-6"
            >
              Education
            </label>
            <input
              id="causes-6"
              name="causes"
              type="checkbox"
              value="Education"
            />
          </div>
          <div>
            <label
              for="causes-7"
            >
              Human Rights
            </label>
            <input
              id="causes-7"
              name="causes"
              type="checkbox"
              value="Human Rights"
            />
          </div>
          <div>
            <label
              for="causes-8"
            >
              Arts
            </label>
            <input
              id="causes-8"
              name="causes"
              type="checkbox"
              value="Arts"
            />
          </div>
          <div>
            <label
              for="causes-9"
            >
              Wildlife and Environment
            </label>
            <input
              id="causes-9"
              name="causes"
              type="checkbox"
              value="Wildlife and Environment"
            />
          </div>
          <div>
            <label
              for="causes-10"
            >
              Elderly
            </label>
            <input
              id="causes-10"
              name="causes"
              type="checkbox"
              value="Elderly"
            />
          </div>
          <div>
            <label
              for="causes-11"
            >
              Poverty
            </label>
            <input
              id="causes-11"
              name="causes"
              type="checkbox"
              value="Poverty"
            />
          </div>
        </fieldset>
        <br />
        <fieldset
          class="EntityCheckboxes"
        >
          <legend>
            Tags
             (all are selected by default)
          </legend>
          <div>
            <label
              for="tags-1"
            >
              virtual
            </label>
            <input
              id="tags-1"
              name="tags"
              type="checkbox"
              value="virtual"
            />
          </div>
          <div>
            <label
              for="tags-2"
            >
              in-person
            </label>
            <input
              id="tags-2"
              name="tags"
              type="checkbox"
              value="in-person"
            />
          </div>
          <div>
            <label
              for="tags-3"
            >
              weekends
            </label>
            <input
              id="tags-3"
              name="tags"
              type="checkbox"
              value="weekends"
            />
          </div>
          <div>
            <label
              for="tags-4"
            >
              weekdays
            </label>
            <input
              id="tags-4"
              name="tags"
              type="checkbox"
              value="weekdays"
            />
          </div>
          <div>
            <label
              for="tags-5"
            >
              nights
            </label>
            <input
              id="tags-5"
              name="tags"
              type="checkbox"
              value="nights"
            />
          </div>
          <div>
            <label
              for="tags-6"
            >
              group
            </label>
            <input
              id="tags-6"
              name="tags"
              type="checkbox"
              value="group"
            />
          </div>
          <div>
            <label
              for="tags-7"
            >
              delivery
            </label>
            <input
              id="tags-7"
              name="tags"
              type="checkbox"
              value="delivery"
            />
          </div>
          <div>
            <label
              for="tags-8"
            >
              cooking
            </label>
            <input
              id="tags-8"
              name="tags"
              type="checkbox"
              value="cooking"
            />
          </div>
          <div>
            <label
              for="tags-9"
            >
              administration
            </label>
            <input
              id="tags-9"
              name="tags"
              type="checkbox"
              value="administration"
            />
          </div>
          <div>
            <label
              for="tags-10"
            >
              cleaning
            </label>
            <input
              id="tags-10"
              name="tags"
              type="checkbox"
              value="cleaning"
            />
          </div>
          <div>
            <label
              for="tags-11"
            >
              heavy lifting
            </label>
            <input
              id="tags-11"
              name="tags"
              type="checkbox"
              value="heavy lifting"
            />
          </div>
          <div>
            <label
              for="tags-12"
            >
              greeting
            </label>
            <input
              id="tags-12"
              name="tags"
              type="checkbox"
              value="greeting"
            />
          </div>
          <div>
            <label
              for="tags-13"
            >
              training
            </label>
            <input
              id="tags-13"
              name="tags"
              type="checkbox"
              value="training"
            />
          </div>
          <div>
            <label
              for="tags-14"
            >
              driving
            </label>
            <input
              id="tags-14"
              name="tags"
              type="checkbox"
              value="driving"
            />
          </div>
          <div>
            <label
              for="tags-15"
            >
              outdoors
            </label>
            <input
              id="tags-15"
              name="tags"
              type="checkbox"
              value="outdoors"
            />
          </div>
        </fieldset>
        <button
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>
</body>
`;
